src/textDocumentProviders.ts

```diff
@@ -94,3 +94,38 @@ export const registerViewAnalyzedCratesProvider = (client: lc.LanguageClient, ct
     }),
   );
 };
+
+export const registerShowSyntaxTreeProvider = (client: lc.LanguageClient, ctx: Context) => {
+  const uri = vscode.Uri.parse("cairo-show-syntax-tree://showSyntaxTree/[SYNTAX_TREE].txt");
+  const eventEmitter = new vscode.EventEmitter<vscode.Uri>();
+
+  const tdcp: vscode.TextDocumentContentProvider = {
+    async provideTextDocumentContent(): Promise<string> {
+      const editor = vscode.window.activeTextEditor;
+      if (!editor) return "No file selected, please click on a file";
+
+      const syntaxTree = await client.sendRequest(showSyntaxTree, {
+        uri: client.code2ProtocolConverter.asUri(editor.document.uri),
+      });
+
+      return syntaxTree ?? "Not available";
```

Comment:
Can't we also do something else here that indicates actual error on the user side?